ChatRPG/Services/EmailSender.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Configuration;
+using Microsoft.AspNetCore.Identity.UI.Services;
+using MailKit.Net.Smtp;
+using MimeKit;
+using MimeKit.Text;
+
+namespace ChatRPG.Services;
+
+public class EmailSender : IEmailSender
+{
+    private readonly string? _senderEmail;
+    private readonly string? _senderPassword;
+    public readonly bool IsActive;
+
+    public EmailSender(IConfiguration configuration)
+    {
+        IsActive = configuration.GetSection("EmailServiceInfo").GetValue<bool>("ShouldSend");
+        if (IsActive)
+        {
+            _senderEmail = configuration.GetSection("EmailServiceInfo").GetValue<string>("Email");
+            _senderPassword = configuration.GetSection("EmailServiceInfo").GetValue<string>("Password");
+
+            if (_senderEmail == null || _senderPassword == null)
+            {
+                throw new ConfigurationErrorsException("Missing 'Email' credentials in appsettings");
+            }
+        }
```

Comment:
Er det ikke bedre kun at tjekke om `configuration.GetSection("EmailServiceInfo").GetValue<bool>("ShouldSend")` er sat i de områder, hvor man laver `EmailSender` objekter? Ellers laves der jo stadig `EmailSender` object, hvorpå man kan kalde dens `public` methods, hvilket virker lidt wack, når `_senderEmail` og `_senderPassword` ikke er sat.

Er det pga du bruger den som en service? For det bør nok kun være en service, hvis den altid skal bruges. Dvs. at der findes en måde at lave mocks af den.